drizzlepac/haputils/photometry_graphics.py

```diff
@@ -86,178 +97,181 @@ def load_data_to_arrays(csv_filename):
+    phot_data['x_index'] = x_index
+    x_index.clear()
 
-    mean_dMagAp1_mean = compute_global_stats(d_MagAp1_mean)
-    mean_dMagAp2_mean = compute_global_stats(d_MagAp2_mean)
+    mean_dMagAp1mean = compute_global_stats(phot_data['Ap1 Mean Differences'])
+    mean_dMagAp2mean = compute_global_stats(phot_data['Ap1 Median Differences'])
 
-    mean_dMagAp1_median = compute_global_stats(d_MagAp1_median)
-    mean_dMagAp2_median = compute_global_stats(d_MagAp2_median)
+    mean_dMagAp1median = compute_global_stats(phot_data['Ap2 Mean Differences'])
+    mean_dMagAp2median = compute_global_stats(phot_data['Ap2 Median Differences'])
 
-    return (d_MagAp1_mean, d_MagAp1_std, d_MagAp1_median), \
-        (d_MagAp2_mean, d_MagAp2_std, d_MagAp2_median), \
-        (mean_dMagAp1_mean, mean_dMagAp1_median), \
-        (mean_dMagAp2_mean, mean_dMagAp2_median)
+    return phot_data, (mean_dMagAp1mean, mean_dMagAp1median), \
+        (mean_dMagAp2mean, mean_dMagAp2median)
 
 
 # Generate the actual plot for the "svm_graphic_type" data.
-def generate_graphic(ap1, ap2, means_ap1, means_ap2):
+def generate_graphic(phot_data, stat_Ap1, stat_Ap2, output_base_filename):
     """Generate the graphics associated with this particular type of data.
 
     Parameters
     ==========
-    ap1: tuple
-    Tuple containing the mean, std, and median statistics for Aperture 1
+    phot_data: Pandas dataframe
+    Dataframe consisting of the Magnitude statistics of mean, std, and median for Aperture 1
+    and Aperture 2
 
-    ap2: tuple
-    Tuple containing the mean, std, and median statistics for Aperture 2
-
-    means_ap1: tuple
+    stat_Ap1: tuple
     Tuple containing the average of the means and the average of the medians for Aperture 1
 
-    means_ap2: tuple
+    stat_Ap2: tuple
     Tuple containing the average of the means and the average of the medians for Aperture 2
+
+    output_base_filename: str
+    Base name for the HMTL file generated by Bokeh.
+
+    Returns
+    =======
+    Nothing
+
+    HTML file is generated and written to the current directory.
     """
 
     # Set the output file immediately as advised by Bokeh.
-    output_file('photometry_grahics.html')
+    output_file(output_base_filename + '.html')
 
-    # Generate a general index array
-    x_index = array.array('i', (i for i in range(0, len(ap1[0]))))
-    num_of_datasets = len(x_index)
+    # Setup the source of the data to be plotted so the axis variables can be
+    # referenced by column name in the Pandas dataframe
+    sourceDF = ColumnDataSource(phot_data)
+    num_of_datasets = len(phot_data.index)
     print('Number of datasets: {}'.format(num_of_datasets))
 
     # Define a figure object
     p1 = figure()
 
     # Add the glyphs
-    p1.circle(x_index, ap1[0], fill_alpha=0.5, line_alpha=0.5, size=10, color='green',
+    p1.circle(x='x_index', y=NEW_PHOT_COLUMNS[0], source=sourceDF, fill_alpha=0.5, line_alpha=0.5, size=10, color='green',
```

Comment:
I found that the Pandas DataFrame automatically provides a column named 'index' which appears to be identical to what you create as 'x_index'.  If so, would it be simpler for your code to just rely on 'index' instead?

Reply:
If I understand your comment, dataframes always have an index, but the index is not actually a column in the data.  I see the "index" on the rows when the file is written/opened as a CSV, but I do not see this when the file is written/opened with HDFStore.  I can follow up on this more, but in the interests of moving on, I will keep "x_index" for now (unless there is further objection).

Reply:
The `ColumnDataSource` object contains a full `index` column as seen here (using `pdb.set_trace()`):

```
(Pdb) fitDF.column_names
['index', 'gen_info.instrument', 'gen_info.detector', 'gen_info.filter', 'gen_info.imgname', 'header.DATE-OBS', 'header.RA_TARG', 'header.DEC_TARG', 'header.GYROMODE', 'header.EXPTIME', 'fit_results.aligned_to', 'fit_results.rms_x', 'fit_results.rms_y', 'fit_results.xsh', 'fit_results.ysh', 'fit_results.rot', 'fit_results.scale', 'fit_results.rot_fit', 'fit_results.scale_fit', 'fit_results.nmatches', 'fit_results.skew', 'x_index']
(Pdb) fitDF.data['index']
array([0, 1, 2])
```
This was what made me think that the `x_index` column was not needed.  If this will not be sufficient, then please keep the `x_index` column as-is.

Reply:
I think the issue here is the difference between a dataframe which has been stored as CSV versus HDF5.  For the CSV file, the dataframe read from the file lists a running index column when just printing the dataframe. However, there is no index column when printing the column names.  Once the a ColumnDataSource object is created from the "CSV dataframe", there is an index column containing what you would expect (e.g, 1, 2, 3, etc.).  This is **not** true when reading the data from an HDF5 file.

When just printing the "HDF dataframe", there is no running index column printed, and there is no index column when printing the column names. There is a column which is printed that contains source file names (e.g., hst_11665_06_wfc3_ir_f110w_ib4606), and this column has no name.  Once the a ColumnDataSource object is created from the "HDF dataframe", there is an index column called 'index', but it contains the source file names and not a running numerical index.  

I checked the HDFStore  to see if there was a parameter on the read of the file to create a numeric index, but this is not the case.